packages/app-builder-lib/src/asar/asarUtil.ts

```diff
@@ -69,95 +77,104 @@ export class AsarPackager {
     }
     await createPackageWithOptions(this.rootForAppFilesWithoutAsar, this.outFile, options)
     console.log = consoleLogger
+  }
 
-    await this.tmpDir.cleanup()
+  private cleanup() {
+    fsNode.rmSync(this.rootForAppFilesWithoutAsar, { recursive: true })
+    this.tmpDir.cleanupSync()
   }
 
   private async detectAndCopy(fileSets: ResolvedFileSet[], cancellationToken: CancellationToken) {
     const taskManager = new AsyncTaskManager(cancellationToken)
     const unpackedPaths = new Set<string>()
     const copiedFiles = new Set<string>()
 
+    const createdSourceDirs = new Set<string>()
+    const links: Array<Link> = []
+    const symlinkType = platform() === "win32" ? "junction" : "file"
```

Comment:
If the symlink points to a directory, would this cause any issues?

Reply:
This is the test case folder tree. Several symlinks are directories.
```
/private/var/folders/y3/r1xf5xzn5pddms2gslqh14sh0000gn/T/et-5a873b0a5efb7affcd771efe4ab31bfa/t-uhAANC/test-project-0/dist/linux-arm64-unpacked/resources/app
├── foo.js -> index.js
├── index.js
├── node_modules
│   ├── debug
│   │   ├── LICENSE
│   │   ├── dist
│   │   │   └── debug.js
│   │   ├── package.json
│   │   └── src
│   │       ├── browser.js
│   │       ├── common.js
│   │       ├── index.js
│   │       └── node.js
│   ├── hello-world
│   │   ├── hello-world-framework.sh
│   │   ├── lib
│   │   │   └── Release
│   │   │       └── Hello.framework
│   │   │           ├── Headers -> Versions/Current/Headers
│   │   │           ├── Hello -> Versions/Current/Hello
│   │   │           ├── Info.plist -> Versions/Current/Info.plist
│   │   │           ├── Modules -> Versions/Current/Modules
│   │   │           └── Versions
│   │   │               ├── A
│   │   │               │   ├── Headers
│   │   │               │   │   └── HelloFramework.h
│   │   │               │   ├── Hello
│   │   │               │   ├── Info.plist
│   │   │               │   └── Modules
│   │   │               │       ├── Hello.abi.json
│   │   │               │       ├── Hello.swiftdoc
│   │   │               │       ├── Hello.swiftmodule
│   │   │               │       │   ├── arm64.private.swiftinterface
│   │   │               │       │   ├── arm64.swiftinterface
│   │   │               │       │   ├── arm64.swiftmodule
│   │   │               │       │   ├── x86_64.private.swiftinterface
│   │   │               │       │   ├── x86_64.swiftinterface
│   │   │               │       │   └── x86_64.swiftmodule
│   │   │               │       ├── Hello.swiftsourceinfo
│   │   │               │       └── module.modulemap
│   │   │               └── Current -> A
│   │   └── package.json
│   └── ms
│       ├── index.js
│       ├── license.md
│       └── package.json
└── package.json
```

Note, I tested this on mac and linux, I'm not sure on windows actually, but I was mirroring this logic here https://github.com/electron-userland/electron-builder/blob/df07101305b24a62731f6b4359228c508c5c76c7/packages/builder-util/src/fs.ts#L298